inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp

```diff
@@ -176,6 +176,116 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
+    auto mult = ngraph::builder::makeEltwise(split->output(1), in2mult, ngraph::helpers::EltwiseTypes::MULTIPLY);
+    auto relu2 = std::make_shared<ngraph::opset1::Relu>(mult);
+
+    auto concat = std::make_shared<ngraph::opset1::Concat>(ngraph::OutputVector{relu1->output(0), relu2->output(0)}, 3);
+    concat->get_output_tensor(0).set_names({"concat"});
+
+    function = std::make_shared<ngraph::Function>(concat, params, "SplitAddConcat");
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorDynamicShapes) {
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = {{1, 5}, {1, 5}, {10, 20}, {10, 20}};
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    OV_ASSERT_NO_THROW(execNet = core->compile_model(function, targetDevice, configuration));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    ov::runtime::Tensor tensor1(ngraph::element::Type_t::f32, {1, 4, 20, 20});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor1));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait_for({}));
+
+    ov::runtime::Tensor tensor2(ngraph::element::Type_t::f32, {1, 4, 20, 20});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor2));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait_for({}));
+
+    ov::runtime::Tensor tensor3(ngraph::element::Type_t::f32, {1, 2, 20, 20});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor3));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait_for({}));
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorGetTensorDynamicShapes) {
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = {{1, 5}, {1, 5}, {10, 20}, {10, 20}};
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    OV_ASSERT_NO_THROW(execNet = core->compile_model(function, targetDevice, configuration));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    ov::runtime::Tensor tensor(ngraph::element::Type_t::f32, {2, 2, 15, 15});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait());
+    ov::runtime::Tensor actual_tensor = req.get_tensor("input_tensor");
+    ASSERT_EQ(tensor.data(), actual_tensor.data());
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorDifferentLayoutsDynamicShapes) {
```

Comment:
Could you also test next scenario (or check that this tests already exist):
1. `get_tensor -> set_shape -> infer -> check results` for input/output
2. tensor with rank > 5 (blocked layout) and different stride: 1) planar - abcdef (should pass) 2) i.e. acdefb (should failed)
3. also we need to cover cases for output blob. 1) set output blob with expected shape after inference -> infer -> and check that output blob contain correct data 2) add the same as `setTensorDifferentLayoutsDynamicShapes` and point 2 for output blob

Reply:
Added tests. Thank you.